include/can/core/dispatch.hpp

```diff
@@ -81,6 +81,46 @@ requires HandlesMessages<HandlerType, MessageTypes...> &&
     Parser<MessageTypes...> parser;
 };
 
+/**
+ * A CanMessageBufferListener for specific nodes that parses messages and
+ * notifies a Handler
+ * @tparam HandlerType A HandlesMessages type
+ * @tparam MessageTypes The Message types this handles
+ */
+template <typename HandlerType, CanMessage... MessageTypes>
+requires HandlesMessages<HandlerType, MessageTypes...> &&
+    (!std::movable<HandlerType> &&
+     !std::copyable<HandlerType>)class DispatchParseTargetNode {
+  public:
+    DispatchParseTargetNode(HandlerType& handler, uint16_t node_id)
+        : handler{handler}, parser{}, node(node_id) {}
+
+    template <bit_utils::ByteIterator Input, typename Limit>
+    requires std::sentinel_for<Limit, Input>
+    void handle(uint32_t arbitration_id, Input input, Limit limit) {
+        if (check_motor(arbitration_id)) {
+            auto arb = ArbitrationId{.id = arbitration_id};
+            auto result = parser.parse(
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+                MessageId{static_cast<uint16_t>(arb.parts.message_id)}, input,
+                limit);
+            handler.handle(result);
+        }
+    }
+
+  private:
+    HandlerType& handler;
+    Parser<MessageTypes...> parser;
+    uint16_t node;
+    auto check_motor(uint32_t arbitration_id) {
```

Comment:
a bit of a nitpick, but can we a) call this `check_node` or something since it could be used for something other than a motor; make the method `const`; and annotate the return type, even though it's not technically necessary
```suggestion
    auto check_node(uint32_t arbitration_id) -> bool const {
```